### PR TITLE
Rewrite Zeek plugin to convert to/from STIX-2 Indicators and Sightings

### PR DESCRIPTION
**Pull request by 0snap**
###  :notebook_with_decorative_cover: Description

Following up on the STIX-2 rewrite of Threat Bus: this PR updates the Zeek plugin.

- Convert STIX-2 Indicators to Zeek/Broker events
- Update Zeek script to consume Zeek/Broker events & ingest in Intel framework
- Convert sightings / matches from Zeek (Broker events) to STIX-2 valid Sightings
- Updated unit tests
- Integration-test the Zeek-plugin's Broker interface in isolation
- Integration-test with a full Zeek instance that runs the Threat Bus app script
- Update / separate from ZMQ message-passing integration test

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/threatbus](https://docs.tenzir.com/threatbus), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Do at least a code-review & run the unit tests. If you want to try out more, here is how:

*Integration testing:*
You require a local Broker installation with Python bindings for integration testing
- Follow the instructions from the Zeek plugin's README
- Run the integration tests

*Interactive usage:*
You require local Zeek and Broker installations with Python bindings for interactive usage
- Make sure your local Zeek and Broker installations have matching versions (i.e., Zeek 4 -> Broker 2.x, Zeek 3.2 -> Broker 1.4)
- Start Threat Bus with the Zeek plugin, in-mem backbone plugin, and the Zmq-app plugin
- Start a local Zeek instance using the Threat Bus app script and monitor your local network interface (`sudo zeek -i wlp0s20f3 -C apps/zeek/threatbus.zeek`)
- Use the test utils to send STIX-2 indicators to the Zmq-endpoint -> that should ultimately be mapped via Broker and arrive in Zeek. Send something like `example.com`.
- Use `curl` or similar to  invoke `example.com` -> Zeek should match that immediately, report a Sighting to Threat Bus and that should be logged as valid STIX-2 (debug logging)

### CODE REVIEW

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py [hunk 1]
```diff
@@ -1,177 +1,169 @@
...
+    "user:user_id": "USER_NAME",
+    "user:account_login": "USER_NAME",
+    "x509-certificate:hashes.'SHA-1'": "CERT_HASH",  # Zeek only supports SHA-1
 }
 
 
-def map_management_message(broker_data, module_namespace):
-    """Maps a management message to an actionable instruction for threatbus.
+def map_management_message(
+    broker_data, module_namespace: str
```
Comment (tobim) [inline plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py:40]: Why does this function not get a handle to the logger? Is the volume of messages too high?

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py [hunk 2]
```diff
@@ -1,177 +1,169 @@
...
-        if not intel_data:
-            return None
-        return broker.zeek.Event(
-            f"{module_namespace}::intel",
-            (msg.ts, str(msg.id), intel_data, msg.operation.value),
+        return None
+
+    # pattern is in the form [file:name = 'foo']
+    (object_path, ioc_value) = indicator.pattern[1:-1].split("=")
+    object_path, ioc_value = object_path.strip(), ioc_value.strip()
```
Comment (tobim) [inline plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py:140]: That syntax threw me a bit off, why not:
```suggestion
    object_path = object_path.strip()
    ioc_value = ioc_value.strip()
```

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py [hunk 3]
```diff
@@ -1,177 +1,169 @@
...
+    zeek_type = zeek_intel_type_map.get(object_path, None)
+    if not zeek_type:
+        logger.debug(
+            f"No matching Zeek type found for STIX-2 indicator type '{object_path}'"
         )
+        return None
+
+    if zeek_type == "URL":
+        # remove leading protocol, if any
+        ioc_value = re.sub(r"^https?://", "", ioc_value)
```
Comment (tobim) [inline plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py:154]: Couldn't this be any protocol, like `ftp://`?

**Review by tobim — COMMENTED** [review]
I tested this locally and can vouch that sightings are logged from threatbus when zeek detects them.

I didn't do a very thorough code review, most of it is in tests which are repetitive by nature. I only have a few suggestions and questions which can be resolved by adding comments.

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py (hunk 3 above)
Comment (0snap) [reply plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py:154]: yes, you are right, we should remove all protocols.

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py (hunk 1 above)
Comment (0snap) [reply plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py]: No the logvolume would be very low. I changed the code to pass a logger instance to that function and made the function log unexpected subscription messages.

File: plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py (hunk 3 above)
Comment (0snap) [reply plugins/apps/threatbus_zeek/threatbus_zeek/message_mapping.py:155]: Added a snip to remove all URL schemes.

**Review by tobim — APPROVED** [review]
Tested again with the latest changes. Everything seems to work as expected.

**Pull request merged**